src/pfe/file-watcher/server/src/projects/ShellExtensionProject.ts

```diff
@@ -204,8 +204,20 @@ export class ShellExtensionProject implements IExtensionProject {
      * @param projectInfo <Required | ProjectInfo> - The metadata information for the project.
      */
     start = async (projectInfo: ProjectInfo): Promise<void> => {
+
         await projectUtil.runScript(projectInfo, path.join(this.fullPath, "entrypoint.sh"), "start");
-        await projectUtil.exposeOverIngress(projectInfo);
+
+        if (process.env.IN_K8) {
+            let port = undefined;
+            const ports = this.getDefaultAppPort();
+            if (Array.isArray(ports)) {
+                port = parseInt(ports[0]);
```

Comment:
If we just pick the first port, does it mean that we may sometimes get the wrong port?

Reply:
For some legacy reasons we allowed https://github.com/eclipse/codewind-appsody-extension/blob/master/.sh-extension#L12 to define multiple app ports before.  But now it only defines 1 app port so it would actually not land here.

Reply:
Would this change impact non-Appsdoy projects that do expose multiple ports?